onchain/rollups/contracts/consensus/quorum/Quorum.sol

```diff
@@ -69,25 +82,99 @@ contract Quorum is AbstractConsensus, PaymentSplitter {
     ///      and the `Quorum` contract must have ownership over
     ///      its current history contract.
     function submitClaim(bytes calldata _claimData) external {
-        // only validators can submit claims
-        uint256 index = validatorIndex[msg.sender];
-        if (index == 0) {
-            revert OnlyValidator();
-        }
+        uint256 id = validatorId[msg.sender];
+        require(id != 0, "Quorum: sender is not validator");
 
-        // If the msg.sender hasn't submitted the same claim before
-        Votes storage claimVotes = votes[_claimData];
-        if (!claimVotes.votedBitMap.get(index)) {
-            claimVotes.votedBitMap.set(index);
+        VotingStatus storage votingStatus = votingStatuses[_claimData];
+        BitMaps.BitMap storage inFavorById = votingStatus.inFavorById;
+
+        if (!inFavorById.get(id)) {
+            // If validator hasn't voted yet, cast their vote
+            inFavorById.set(id);
 
             // If this claim has now just over half of the quorum's votes,
             // then we can submit it to the history contract.
-            if (++claimVotes.count == 1 + numOfValidators / 2) {
+            if (++votingStatus.inFavorCount == 1 + numOfValidators / 2) {
                 history.submitClaim(_claimData);
             }
         }
     }
 
+    /// @notice Get an array with the IDs of all validators.
+    /// @return Array of IDs of validators
+    function validatorIds() external view returns (uint256[] memory) {
```

Comment:
What's the use case for getting an array with ids? Is it sufficient to get `numOfValidators` instead?

Reply:
Yes, we say in the documentation that IDs range from 1 to `numOfValidators`.
Maybe we should remove this function. :-)